lib/keystore.ex

```diff
@@ -23,11 +23,13 @@ defmodule Keystore do
     validate_empty_path!(decoded_json["path"])
 
     privkey = decrypt!(decoded_json["crypto"], password)
-    # TODO: derive from privkey and validate with this pubkey
+
     pubkey = Map.fetch!(decoded_json, "pubkey") |> parse_binary!()
-    if Bls.derive_pubkey(privkey) == pubkey do
+
+    if Bls.derive_pubkey(privkey) != {:ok, pubkey} do
```

Comment:
Let's match against the `:ok` outside the `if`, so we differentiate between pubkey deriving errors and "pubkeys differ" errors:

```suggestion
    {:ok, derived_pubkey} = Bls.derive_pubkey(privkey)

    if derived_pubkey != pubkey do
```